### ISSUE TITLE
Support for Umbraco 15

### ISSUE THREAD

**Issue by EddieDemon** [Shazwazza/Articulate#451]
I just updated to Umbraco 14 and noticed an error which did not exist before Umbraco 14. Now this error only shows itself when I add Articulate (truth being it's the only additional package I added.) I'm unsure why this happens, since you always have `Umbraco.Cms.Api.Delivery` and `Umbraco.Cms.Api.Management`. Only when Articulate is added it becomes an issue.

I'm some what sure it's me, but I would love someone else to look over this.

`error CS0121: The call is ambiguous between the following methods or properties: 'Umbraco.Extensions.UmbracoBuilderExtensions.AddBackOffice(Umbrac
o.Cms.Core.DependencyInjection.IUmbracoBuilder, System.Action<Microsoft.Extensions.DependencyInjection.IMvcBuilder>?)' and 'Umbraco.Extensions.UmbracoBuilderExtensions.AddBackOffice(Umbraco.Cms.Core.Dependenc
yInjection.IUmbracoBuilder, System.Action<Microsoft.Extensions.DependencyInjection.IMvcBuilder>?)'`

Now, to reproduce:
```powershell
dotnet new install Umbraco.Templates
dotnet new umbraco
dotnet add package Articulate
dotnet build
```

**Comment by gavinfaux** [comment]
'Umbraco.Cms.Web.Backoffice' was deprecated for version 14 as was Smidge, v5 not compatible with Umbraco 14 currently.

![451a](https://github.com/user-attachments/assets/58569be1-ad6d-4933-a685-44942d9c113e)

![451b](https://github.com/user-attachments/assets/dbdfa446-7283-4c36-8da9-e938cb1e0181)

**Comment by Shazwazza** [comment]
Yeah, I've not tried Smidge with v14. It clearly needs some updates and another major version.

**Comment by Shazwazza** [comment]
There's plenty of UI in Articulate for the back office so that will need a big overhaul and I have no experience doing any UI stuff for Umbraco 14. If anyone wants to help migrate the codebase to work, that would be wonderful. Unfortunately it took a very long time to get the last update working with Umbraco 10, this one for UI might take a while too :(

**Comment by gavinfaux** [comment]
> There's plenty of UI in Articulate for the back office so that will need a big overhaul and I have no experience doing any UI stuff for Umbraco 14. If anyone wants to help migrate the codebase to work, that would be wonderful. Unfortunately it took a very long time to get the last update working with Umbraco 10, this one for UI might take a while too :(

When I last looked at this backend of 2024 I did not have any issue with updating core, Management API's and Themes amends, but encountered issues with basic routing (might have been something I missed or mistakes), so did not proceed with reworked back office.

I have some time on my hands, so happy to get back in to this and for the Umbraco 14/15+ version, is there anything already done on this I can assist with else I can get started on with a new umbraco-extension package?

**Comment by Shazwazza** [comment]
@gavinfaux yes!!!! That would be amazing. Basically most things all work, it's just there's a lot is UI components in articulate for the back office that all need to be migrated to the new back office UI framework. There's a few other features that haven't been fully ported over to v13+ but the main thing preventing compact with v14+ is the UI stuff. 

If you know the magic sauce to get that done or even make a start, that would be absolutely fantastic 🎉

**Comment by JohnBergman** [comment]
What do you think about replacing the markdown editor package you added with the built in one in v16?

https://docs.umbraco.com/umbraco-cms/fundamentals/backoffice/property-editors/built-in-umbraco-property-editors/markdown-editor

or perhaps

https://marketplace.umbraco.com/package/jcdcdev.umbraco.extendedmarkdowneditor

Then the only custom control seems like it would be the theme picker.

**Comment by Shazwazza** [comment]
The reason there was a custom md editor was to fix the old broken one. More than happy to use the new/existing one if it works.

**Comment by Shazwazza** [comment]
Hi all, I've updated the title here to be more specific. We don't really care about v14 right now, but v15 we do.

I've pushed a branch/PR which does 'build' but needs lots of work, see https://github.com/Shazwazza/Articulate/pull/455

Any help is much appreciated, see the info in that PR

**Comment by moriyama-nik** [comment]
@Shazwazza / @gavinfaux  I was just wondering if there was any update on the V17 support (I know this thread is for v15). When I was testing it before the PR merged it seemed to be working well for me. Even if it's a public beta version? :-)

**Comment by Shazwazza** [comment]
@moriyama-nik lets ship this thing!! I'll try to find time today - just swamped 😅

**Comment by Shazwazza** [comment]
@gavinfaux + @moriyama-nik here's an RC release: https://www.nuget.org/packages/Articulate/6.0.0-rc.2 

Happy to ship RTM if anyone can confirm the install, etc... works

**Comment by moriyama-nik** [comment]
Thanks @Shazwazza that's great news!
